src/transformers/models/seggpt/modeling_seggpt.py

```diff
@@ -0,0 +1,939 @@
+    tensor = tensor.reshape(shape=(batch_size, 3, patch_height * patch_size, patch_width * patch_size))
+
+    return tensor
+
+
+@dataclass
+class SegGptEncoderOutput(BaseModelOutput):
+    """
+    Output type of [*SegGptEncoderOutput*].
+    Args:
+        last_hidden_state (*torch.FloatTensor* of shape *(batch_size, patch_height, patch_width, hidden_size)*):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (*Tuple[torch.FloatTensor]*, *optional*, returned when `config.output_hidden_states=True`):
+            Tuple of *torch.FloatTensor* (one for the output of the embeddings + one for the output of each layer)
+            of shape *(batch_size, patch_height, patch_width, hidden_size)*.
+        attentions (*Tuple[torch.FloatTensor]*, *optional*, returned when `config.output_attentions=True`):
+            Tuple of *torch.FloatTensor* (one for each layer) of shape
+            *(batch_size, num_heads, seq_len, seq_len)*.
+        intermediate_features (*Tuple[torch.FloatTensor]*, *optional*, returned when `config.encoder_output_indicies` is set):
+            Tuple of *torch.FloatTensor* of shape *(batch_size, patch_height, patch_width, hidden_size)*.
+            Each element in the Tuple corresponds to the output of the layer specified in `config.encoder_output_indicies`.
+            Additionaly, each feature passes through a LayerNorm.
+    """
+
+    intermediate_features: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class SegGptImageSegmentationOutput(ModelOutput):
+    """
+    Output type of [*SegGptImageSegmentationOutput*].
+
+    Args:
+        pred_masks (*torch.FloatTensor* of shape *(batch_size, num_channels, height, width)*):
+            The predicted masks.
+        hidden_states (*Tuple[torch.FloatTensor]*, *optional*, returned when `config.output_hidden_states=True`):
+            Tuple of *torch.FloatTensor* (one for the output of the embeddings + one for the output of each layer)
+            of shape *(batch_size, patch_height, patch_width, hidden_size)*.
+        attentions (*Tuple[torch.FloatTensor]*, *optional*, returned when `config.output_attentions=True`):
+            Tuple of *torch.FloatTensor* (one for each layer) of shape
+            *(batch_size, num_heads, seq_len, seq_len)*.
+        loss (*torch.FloatTensor*, *optional*, returned when `labels` is provided):
+            The loss value.
+
+    """
+
+    pred_masks: torch.FloatTensor
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+    loss: Optional[torch.FloatTensor] = None
+
+
+class SegGptEmbeddings(nn.Module):
+    """
+    Construct the embeddings from patch, position embeddings for input and prompt.
+    """
+
+    def __init__(self, config: SegGptConfig) -> None:
+        super().__init__()
+
+        self.mask_token = nn.Parameter(torch.zeros(1, 1, 1, config.hidden_size))
```

Comment:
this parameter is also not useful, only used during masked image pre-training, so can be removed

Reply:
But it is used in the `SegGptEmbeddings` forward

```python
if bool_masked_pos is not None:
    mask_token = self.mask_token.expand(batch_size, patch_height, patch_width, -1)
    # replace the masked visual tokens by mask_token
    w = bool_masked_pos.unsqueeze(-1).type_as(mask_token).reshape(-1, patch_height, patch_width, 1)
    prompt_embeddings = prompt_embeddings * (1 - w) + mask_token * w
```

And since `bool_masked_pos` always defaults to something other than `None` something is added to the `prompt_embeddings`. Or am I missing something?

Reply:
So the `mask_token` parameteter is effectively used at inference time?